sense_hat_example/sense_stick.js

```diff
@@ -0,0 +1,385 @@
+            return Sk.importMainWithBody("<stdin>", false, editor.getValue(), true);
+        }, {'*': interruptHandler});
+
+        // Bind key events
+        document.body.addEventListener('keydown', handleKeyDown);
+        document.body.addEventListener('keypress', handleKeyPress);
+        document.body.addEventListener('keyup', handleKeyUp);
+
+        pr.then(function () {
+            output.value = output.value + "\nFinished";
+
+            // Remove any listenerer to avoid weird side effects, !IMPORTANT!
+            cleanAfterRun();
+        }, function (err) {
+            console.info('errorHandler', err);
+            if (err.nativeError && err.nativeError.message === 'KeyboardInterrupt') {
+                output.value = output.value + "\nStopped!";
+            } else {
+                console.error(err);
+                output.value = output.value + "\n" + err.toString();
+            }
+
+            // Remove any listenerer to avoid weird side effects, !IMPORTANT!
+            cleanAfterRun();
+        });
+    });
+}
+
+/**
+ * jQuery based Event Emitter Interface, no need to pull in a EventEmitter lib
+ */
+(function(jQuery) {
+ 
+  jQuery.eventEmitter = {
+    _JQInit: function() {
+      this._JQ = jQuery(this);
+    },
+    emit: function(evt, data) {
+      !this._JQ && this._JQInit();
+      this._JQ.trigger(evt, data);
+    },
+    once: function(evt, handler) {
+      !this._JQ && this._JQInit();
+      this._JQ.one(evt, handler);
+    },
+    on: function(evt, handler) {
+      !this._JQ && this._JQInit();
+      this._JQ.bind(evt, handler);
+    },
+    off: function(evt, handler) {
+      !this._JQ && this._JQInit();
+      this._JQ.unbind(evt, handler);
+    }
+  };
+ 
+}($));
+
+/**
+ * Handles the sense stick events
+ */
+function SenseStickDevice(sendSignalToSkulpt) {
+    this._eventListeners;
+    this._eventQueue = [];
+    this._threadHandler = null;
+    this._isDownDict = {};
+    this.sendSignalToSkulpt = sendSignalToSkulpt;
+}
+
+// inheritance, or so :P
+$.extend(SenseStickDevice.prototype, jQuery.eventEmitter);
+
+SenseStickDevice.prototype.triggerKeyboardInterrupt = function () {
+    this.emit('sensestick.input', { type: 'keyboardinterrupt'});
+}
+
+SenseStickDevice.prototype.addKeyDownEventToDict = function (keyStr) {
+    this._isDownDict[keyStr] = true;
+}
+
+SenseStickDevice.prototype.removeKeyDownEventFromDict = function (keyStr) {
+    delete this._isDownDict[keyStr];
+}
+
+SenseStickDevice.prototype.isKeyDown = function (keyStr) {
+    return this._isDownDict[keyStr] === true;
+}
+
+SenseStickDevice.prototype.push = function (key, state, type) {
+    var event = {
+        type: type != null ? parseInt(type) : SenseStickDevice.EV_KEY,
+        key: parseInt(key),
+        state: parseInt(state),
+        timestamp: Date.now()
+    };
+
+    this._enqueue(event);
+
+    var shouldTriggerSignal = false;
+    if (Sk.sense_hat.sensestick._threadHandler != null) {
```

Comment:
Check if there is an active _threadHandler, and if `true` we need to send a signal to the process.